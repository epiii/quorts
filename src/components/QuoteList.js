import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import { format } from "date-fns";

import Quote from "./Quote";

const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 16px;
`;

const GridItem = styled.div`
  margin-bottom: 16px;
`;

const QuoteList = ({ items, onClickCopy, onClickRemove }) => (
  <GridContainer>
    {[...Array(3)].map((_, index) => (
      <div key={index}>
        {items.filter((_, itemIndex) => itemIndex % 3 === index).map(item => (
          <GridItem key={item.id}>
            <Quote
              text={item.text}
              date={format(item.date, "MMM DD, YYYY - HH:mm A")}
              onClickRemove={() => onClickRemove(item.id)}
              onClickCopy={() => onClickCopy(item.text)}
            />
          </GridItem>
        ))}
      </div>
    ))}
  </GridContainer>
);

QuoteList.propTypes = {
  items: PropTypes.array,
  onClickCopy: PropTypes.func,
  onClickRemove: PropTypes.func
};

QuoteList.defaultProps = {
  items: []
};

export default QuoteList;
