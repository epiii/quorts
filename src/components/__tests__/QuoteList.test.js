import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import QuoteList from "../QuoteList";

describe("QuoteList", () => {
  Date.now = jest.fn(() => 1529122840652);

  const setup = propOverrides => {
    const props = {
      items: [
        {
          id: "1",
          text: "My Quote 1",
          date: Date.now()
        },
        {
          id: "2",
          text: "My Quote 2",
          date: Date.now()
        },
        {
          id: "3",
          text: "My Quote 3",
          date: Date.now()
        }
      ],
      onClickRemove: jest.fn(),
      onClickCopy: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<QuoteList {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
