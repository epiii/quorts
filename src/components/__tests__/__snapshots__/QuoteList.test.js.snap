// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteList should renders default 1`] = `
<styled.div>
  <div
    key="0"
  >
    <styled.div
      key="1"
    >
      <Quote
        date="Jun 16, 2018 - 11:20 AM"
        onClickCopy={[Function]}
        onClickRemove={[Function]}
        text="My Quote 1"
      />
    </styled.div>
  </div>
  <div
    key="1"
  >
    <styled.div
      key="2"
    >
      <Quote
        date="Jun 16, 2018 - 11:20 AM"
        onClickCopy={[Function]}
        onClickRemove={[Function]}
        text="My Quote 2"
      />
    </styled.div>
  </div>
  <div
    key="2"
  >
    <styled.div
      key="3"
    >
      <Quote
        date="Jun 16, 2018 - 11:20 AM"
        onClickCopy={[Function]}
        onClickRemove={[Function]}
        text="My Quote 3"
      />
    </styled.div>
  </div>
</styled.div>
`;
